### ISSUE TITLE
[SGD] Document best practices for Pipeline epochs

### ISSUE THREAD

**Issue by matthewdeng** [ray-project/ray#19323]
There are a few reasonable approaches that are "okay". Tracking this issue here to update documentation to clarify this when a stronger opinion is formed.

Options:
1. Define in training function - more "natural" to the user.
2. Define in `repeat(num_epochs)` - avoids computation of `num_epochs+1`th epoch (which will never get read).
3. Both (1) and (2)?

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
